### PR TITLE
refactor: add proxy parsing for fb app events

### PR DESCRIPTION
**Pull request by crownedlake**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by saikumarrs** [comment]
This PR should be a "feat" rather than "refactor".

File: v0/constants/fbMarketApi.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    };
+  }
+
+  if (THROTTLED_ERROR_CODES.includes(error.code)) {
+    errorStatus = 429;
+    statTags = {
+      ...statTags,
+      meta: getDynamicMeta(errorStatus)
+    };
+  }
```
Comment (saikumarrs) [inline v0/constants/fbMarketApi.js:51]: All these blocks should be mutually exclusive. Any mistake in the mapping could result in erroneous behavior.

File: v0/destinations/fb/network-handler.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    // successful response from facebook app event api
+    return;
+  }
+  const { error } = response;
+  const statusAndStats = getStatusAndStats(error, DESTINATION);
+  throw new ErrorBuilder()
+    .setStatus(statusAndStats.status)
+    .setDestinationResponse({ ...response, status: destResponse.status })
+    .setMessage(
+      `Facebook App Event: Failed with ${error.message} during response transformation`
```
Comment (saikumarrs) [inline v0/destinations/fb/network-handler.js:28]:
```suggestion
      `Facebook App Events: Failed with "${error.message}" during response transformation`
```

File: v0/destinations/fb/transformErrors.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+const statTags = destination => ({
+  destType: destination,
+  stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+  scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE
+});
+
+const INVALID_NAME_ERROR = (eventName, pattern) =>
+  new ErrorBuilder()
+    .setMessage(
+      `Event name ${eventName} is not a valid FB APP event name.It must match the regex ${pattern}`
```
[the remark this answers is not in the document]
Comment (crownedlake) [reply v0/destinations/fb/transformErrors.js:13]: @saikumarrs I just wrapped the earlier error messages. However your suggestion makes sense, will fix it.

File: v0/destinations/fb/transform.js [hunk 4]
```diff
@@ -352,7 +346,7 @@ function processSingleMessage(message, destination) {
       break;
     default:
       logger.error("could not determine type");
-      throw new CustomError("message type not supported", 400);
+      throw UNSUPPORTED_TYPE_ERROR;
```
[the remark this answers is not in the document]
Comment (crownedlake) [reply v0/destinations/fb/transform.js:349]: Will fix this.

File: v0/destinations/fb/config.js [hunk 5]
```diff
@@ -28,5 +28,6 @@ module.exports = {
   eventNameMapping,
   eventPropsMapping,
   eventPropsToPathMapping,
-  eventPropToTypeMapping
+  eventPropToTypeMapping,
+  DESTINATION: "FACEBOOK_APP_EVENT"
```
[the remark this answers is not in the document]
Comment (crownedlake) [reply v0/destinations/fb/config.js:32]: Will recheck this.

File: v0/constants/fbMarketApi.js (hunk 1 above)
Comment (crownedlake) [reply v0/constants/fbMarketApi.js:51]: @saikumarrs I have moved this code to a commonplace from `v0/destinations/facebook_pixel/network-handler.js`. I will fix this. @sanpj2292 should I change at both places?

File: v0/destinations/fb/network-handler.js (hunk 2 above)
Comment (crownedlake) [reply v0/destinations/fb/network-handler.js:28]: fixed.

**Pull request closed without merging**